### PR TITLE
feat(proxy): add cache

### LINKED ISSUE #326: feat(proxy): add customizable caching

### PR DESCRIPTION
**Pull request by kckeiks**
## Why

Adds cache to proxy.

Fixes #326 

## Demo

 
```
Summary:
  Success rate: 1.0000
  Total:        2.5374 secs
  Slowest:      0.0985 secs
  Fastest:      0.0002 secs
  Average:      0.0013 secs
  Requests/sec: 39411.1637

  Total data:   2.52 GiB
  Size/request: 26.41 KiB
  Size/sec:     1016.46 MiB

Response time histogram:
  0.000 [1]     |
  0.010 [99844] |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■
  0.020 [95]    |
  0.030 [10]    |
  0.040 [0]     |
  0.049 [2]     |
  0.059 [11]    |
  0.069 [10]    |
  0.079 [10]    |
  0.089 [9]     |
  0.099 [8]     |

Latency distribution:
  10% in 0.0006 secs
  25% in 0.0009 secs
  50% in 0.0011 secs
  75% in 0.0015 secs
  90% in 0.0018 secs
  95% in 0.0021 secs
  99% in 0.0039 secs

Details (average, fastest, slowest):
  DNS+dialup:   0.0402 secs, 0.0341 secs, 0.0897 secs
  DNS-lookup:   0.0000 secs, 0.0000 secs, 0.0002 secs

Status code distribution:
  [200] 100000 responses
```

## Checklist

- N/A I have made corresponding changes to the tests
- N/A I have made corresponding changes to the documentation
- [x] I have run the app using my feature and ensured that no functionality is broken

### CODE REVIEW

File: crates/ursa-proxy/Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "ursa-proxy"
-authors = ["Miguel Guarniz <miguel@fleek.xyz>"]
+authors = ["Miguel Guarniz <miguel@fleek.xyz>", "Scott <scott@fleek.xyz>"]
```
[the remark this answers is not in the document]
Comment (b0xtch) [reply crates/ursa-proxy/Cargo.toml:3]: We have info@fleek.network, but we create something cooler like code@fleek.network

**Review by b0xtch — DISMISSED** [review]

**Comment by b0xtch** [comment]
Mint

File: crates/ursa-proxy/Cargo.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (kckeiks) [reply crates/ursa-proxy/Cargo.toml:3]: dev@fleek.network?

File: crates/ursa-proxy/src/main.rs [hunk 2]
```diff
@@ -1,22 +1,25 @@
...
 async fn main() -> Result<()> {
     let Cli {
         command: Commands::Daemon(opts),
     } = Cli::parse();
     let config = load_config(&opts.config.parse::<PathBuf>()?)?;
-    ProxyCore::new(config).start_servers().await.unwrap();
+    let moka_config = config.moka.clone().unwrap_or_default();
+    let cache = MokaCache::new(moka_config.stream_buf);
+    Proxy::new(config, cache.clone()).start().await.unwrap();
     Ok(())
```
Comment (ozwaldorf) [inline crates/ursa-proxy/src/main.rs:24]:
```suggestion
    Proxy::new(config, cache.clone()).start().await
```

File: crates/ursa-proxy/Cargo.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (ozwaldorf) [reply crates/ursa-proxy/Cargo.toml:3]: maybe could resolve separately, across the entire project, once we decide/setup the email?

**Review by scott-dn — APPROVED** [review]

**Review by ozwaldorf — APPROVED** [review]

**Pull request merged**